javalib/src/main/scala/java/util/ArrayList.scala

```diff
@@ -0,0 +1,80 @@
+package java.util
+
+import scala.scalajs._
+import scala.collection.JavaConversions._
+
+class ArrayList[E] private (inner: js.Array[Box[E]]) extends AbstractList[E]
+                                                        with RandomAccess
+                                                        with Cloneable
+                                                        with Serializable {
+  self =>
+
+  def this(initialCapacity: Int) = {
+    this(new js.Array[Box[E]])
+    if (initialCapacity < 0)
+      throw new IllegalArgumentException
+  }
+
+  def this() =
+    this(10)
+
+  def this(c: Collection[_ <: E]) = {
+    this(c.size)
+    addAll(c)
+  }
+
+  def trimToSize(): Unit = {
+    // We ignore this as js.Array doesn't support explicit pre-allocation
+  }
+
+  def ensureCapacity(minCapacity: Int): Unit = {
+    // We ignore this as js.Array doesn't support explicit pre-allocation
+  }
+
+  def size(): Int =
+    inner.length
+
+  override def clone(): AnyRef =
+    new ArrayList(inner.jsSlice(0))
+
+  def get(index: Int): E = {
+    checkIndexInBounds(index)
+    inner(index).inner
+  }
+
+  override def set(index: Int, element: E): E = {
+    val e = get(index)
+    inner(index) = Box(element)
+    e
+  }
+
+  override def add(e: E): Boolean = {
+    inner += Box(e)
+    true
+  }
+
+  override def add(index: Int, element: E): Unit = {
+    checkIndexOnBounds(index)
+    inner.insert(index, Box(element))
+  }
+
+  override def remove(index: Int): E = {
+    checkIndexInBounds(index)
+    inner.remove(index).inner
+  }
+
+  override def clear(): Unit =
+    inner.clear()
+
+  override def addAll(index: Int, c: Collection[_ <: E]): Boolean = {
+    checkIndexOnBounds(index)
+    val mod = c.nonEmpty
+    if (mod)
+      inner.splice(index, 0, c.map(Box(_)).toArray: _*)
+    mod
```

Comment:
I'm pretty sure this is worse than the inherited implementation ^^

Either don't override `addAll` at all, or else pattern-match `c` for an `ArrayList`, and access its `inner` directly to call `splice`. But if `c` is not an `ArrayList`, I think the inherited implementation is better.